pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -607,6 +578,54 @@ func (oe *operationExecutor) generateDetachVolumeFunc(
 	}, nil
 }
 
+func (oe *operationExecutor) verifyVolumeIsSafeToDetach(
+	volumeToDetach AttachedVolume) error {
+	// Fetch current node object
+	node, fetchErr := oe.kubeClient.Core().Nodes().Get(volumeToDetach.NodeName)
+	if fetchErr != nil {
+		if errors.IsNotFound(fetchErr) {
+			glog.Warningf("Node %q not found on API server. DetachVolume will skip safe to detach check.",
+				volumeToDetach.NodeName,
+				volumeToDetach.VolumeName,
+				volumeToDetach.VolumeSpec.Name())
+			return nil
```

Comment:
By returning nil here, it can unblock DetachVolume in operation_executor. But from our current workflow, it is not clear to me how DetachVolume can block AttachVolume. No matter detach fails or not, reconciler will try to attach the volume through gce service. The following cases might happen
1. If the volume is still attached by another node, attach will fail. 
2. If the node to which the volume is originally attached is deleted, attach will succeed (I think no matter detach failed or not)
3. If the node to which the volume is originally attached is removed from API server (by kubectl delete), attach failed at first but succeed eventually    because volume is detached after timeout (waiting for unmount status update).
So it is not clear to me that in the issue #29903, during upgrading, if the old node is deleted, the following attach to the new node should succeed. (from the kubelet log I checked, kubelet verifyVolumeAttached succeed because the volume is in the attached list of the new node), but failed on checking the devicePath before trying to mount volume.

So I just want to make sure we cover all the cases.